frontend/src/views/AccountPages/PlansBuyPage/PlansBuyPage.js

```diff
@@ -42,7 +41,24 @@ export default function PlanBuyPage(props) {
 
     const planActive = useSelector(state => state.planActive);
     const { loading, plans, error } = planActive;
+    const userSignin = useSelector(state => state.userSignin);
+    const { userInfo } = userSignin;
     const dispatch = useDispatch();
+    let yearDate = new Date();
+    yearDate = yearDate.setFullYear(yearDate.getFullYear() + 1);
+
+    function checkDate(planDate, yearDate, duration) {
+        console.log(planDate)
+        planDate.setMonth(planDate.getMonth() + duration);
```

Comment:
Avoid mutating any parameter that's passed in, unless that's the point of the function to modify it. It can lead to odd behaviour as the function is a `query` type function.